research-app/src/App.vue

```diff
@@ -2195,12 +2198,18 @@ export default class App extends WWTAwareComponent {
     const rowSeparator = "\r\n";
     const colSeparator = "\t";
 
-    for (const catalog of this.visibleHipsCatalogs()) {
-      const catalogName = catalog.name;
-      const layer = this.layerForHipsCatalog(catalogName);
-      if (layer == null) {
-        continue;
+    for (const layerInfo of this.visibleTableLayers()) {
+
+      let layer: SpreadSheetLayer | null = null;
+      if (isImageSetInfo(layerInfo)) {
+        layer = this.layerForHipsCatalog(layerInfo.name);
+      } else {
+        layer = this.spreadSheetLayerById(layerInfo.id);
```

Comment:
*Maybe* it would be worth centralizing this logic somewhere? I feel like the code that handles the different enum-type cases should be found near the type ... if we, say, add a new option to the enum one day, we'd have to touch code in several different places to update it all to handle the new type.